xline-client/src/clients/lease.rs

```diff
@@ -39,6 +53,117 @@ impl LeaseClient {
                 token.as_ref().and_then(|t| t.parse().ok().map(Arc::new)),
             )),
             token,
+            id_gen,
+        }
+    }
+
+    /// Send `LeaseGrantRequest` by `CurpClient`
+    ///
+    /// # Errors
+    ///
+    /// If `CurpClient` failed to send request
+    #[inline]
+    pub async fn grant(&mut self, mut request: LeaseGrantRequest) -> Result<LeaseGrantResponse> {
+        let propose_id = self.generate_propose_id();
+        if request.inner.id == 0 {
+            request.inner.id = self.id_gen.next();
         }
+        let request = RequestWithToken::new_with_token(
+            xlineapi::LeaseGrantRequest::from(request).into(),
+            self.token.clone(),
+        );
+        let cmd = Command::new(vec![], request, propose_id);
+        let cmd_res = self.curp_client.propose(cmd).await?;
+        Ok(cmd_res.decode().into())
+    }
+
+    /// Send `LeaseRevokeRequest` by `CurpClient`
+    ///
+    /// # Errors
+    ///
+    /// If `CurpClient` failed to send request
+    #[inline]
+    pub async fn revoke(&mut self, request: LeaseRevokeRequest) -> Result<LeaseRevokeResponse> {
+        let propose_id = self.generate_propose_id();
+        let request = RequestWithToken::new_with_token(
+            xlineapi::LeaseRevokeRequest::from(request).into(),
+            self.token.clone(),
+        );
+        let cmd = Command::new(vec![], request, propose_id);
```

Comment:
Lease revoke request should conflict with other requests that have the same keys attached to this lease, but we cannot get keys at client side, maybe we need to let this request send to `LeaseServer`

Reply:
done